apps/institution-send-mail-scheduler/src/main/java/it/pagopa/selfcare/institution/service/InstitutionSendMailScheduledServiceImpl.java

```diff
@@ -0,0 +1,106 @@
+                                                   @ConfigProperty(name = "institution-send-mail.notification-start-date") String startDate,
+                                                   ProductService productService,
+                                                   @ConfigProperty(name = "institution-send-mail.notification-query-size") Integer querySize) {
+        this.mailService = mailService;
+        this.templateMail = templateMail;
+        this.productService = productService;
+        this.startDate = startDate;
+        this.sendingFrequency = sendingFrequency;
+        this.querySize = querySize;
+    }
+
+    @Override
+    public Uni<Integer> retrieveInstitutionFromPecNotificationAndSendMail() {
+        Long moduleDayOfTheEpoch = calculateModuleDayOfTheEpoch();
+        int startPage = 0;
+        return retrieveFilteredAndPaginatedPecNotification(moduleDayOfTheEpoch, startPage, querySize);
+    }
+
+    public Uni<Integer> retrieveFilteredAndPaginatedPecNotification(Long moduleDayOfTheEpoch, int page, int size) {
+        log.info("Retrieving page " + page + " of PecNotification");
+
+        var pecNotificationPage = PecNotification.find(PecNotification.Fields.moduleDayOfTheEpoch.name(), moduleDayOfTheEpoch)
+                .page(page, size);
+
+        return Uni.combine().all()
+                .unis(pecNotificationPage.hasNextPage(), pecNotificationPage.list())
+                .asTuple()
+                .onItem().transformToUni(tuple -> {
+                    if(tuple.getItem1()) {
+                        return retrievePecNotificationListAndSendMail(tuple.getItem2())
+                                .onItem().transformToUni(unused -> retrieveFilteredAndPaginatedPecNotification(moduleDayOfTheEpoch, page + 1, size))
+                                .replaceWith(0);
+                    }else{
+                        return retrievePecNotificationListAndSendMail(tuple.getItem2())
+                                .replaceWith(0);
+                    }
+                })
+                .onFailure().invoke(throwable -> log.error("Error during send scheduled mail", throwable));
+
+    }
+
+    private Uni<Void> retrievePecNotificationListAndSendMail(List<ReactivePanacheMongoEntityBase> query) {
+        return Multi.createFrom().iterable(query)
+                .onItem().transform(entityBase -> (PecNotification) entityBase)
+                .onItem().transformToUniAndMerge(this::constructAndSendMail)
+                .onFailure().invoke(throwable -> log.error("Error during send scheduled mail", throwable))
+                .collect().asList()
+                .replaceWith(Uni.createFrom().voidItem());
+    }
+
+    private Uni<Void> constructAndSendMail(PecNotification pecNotification) {
+        Product product = productService.getProduct(pecNotification.getProductId());
+        Map<String, String> mailParameters = retrieveMailParameters(product.getTitle());
```

Comment:
you could use `Map.of(PRODUCT_PLACEHOLDER, product.getTitle())` removing method **retrieveMailParameters**